pallets/msa/src/rpc/src/lib.rs

```diff
@@ -72,6 +73,17 @@ pub trait MsaApi<BlockHash, AccountId> {
 		&self,
 		msa_id: MessageSourceId,
 	) -> RpcResult<Option<KeyInfoResponse<AccountId>>>;
+
+	/// Retrieve a provider's application context or default provider context
+	/// If application id is not specified, rpc will return the default provider context
+	/// The default provider context is registered at time of provider registration.
+	#[method(name = "msa_getProviderApplicationContext")]
```

Comment:
I think we had a consensus to not add new RPCs (only runtimes) unless there is a functionality that we can't do solely from inside runtime. This would simplify the upgrades and keep our node code simple. Please let me know if there are questions or concerns.

Reply:
Good point, i will see how Provider Dashboard is using the provider state queries right now and if they are state queries we can get rid of RPC itself

Reply:
Ok makes me to not have RPC, i have removed the RPC and retained the runtime api for this and updated e2e tests